packages/block-library/src/footnotes/index.js

```diff
@@ -21,9 +21,8 @@ export const settings = {
 	edit,
 };
 
-// Would be good to remove the format and HoR if the block is unregistered.
-registerFormatType( formatName, format );
-
 export const init = () => {
+	// Would be good to remove the format and HoR if the block is unregistered.
+	registerFormatType( formatName, format );
```

Comment:
Regarding the comment above, this does register the format when the block is registered, but I feel like it should be tied closer to block type registration AND reregistration since these two depend on each other. Any ideas?

Reply:
Maybe we can use side-effect and `blocks.registerBlockType` register format when footnotes. But I don't see a corresponding unregister filter in the blocks package.

Reply:
What do folks think? Is the absence of a `blocks.deregisterBlockType` hook a blocker?